docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+# Practical Example - Birthday Kata
```

Comment:
I think many folks may not know what the birthday Kata is off-hand. What if we call this Practical Example - Modeling a Birthday Message" or something like that, and then in the very next section, under "# The Kata" we can both link to the blog and foreground it with some context to help non DDD aficionados come up to speed.